ex4B/ex4.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+function toNumber(str) {
+
+	var strNum = '', strDec = '', number = null, i = 0, allowDot = true, negative = false;
+
+	// Negative sige can appear only at the start of string and followed by number
+	//if (str.slice(0, 1) === '-' && isNumber(str.slice(1, 2))) {
+	
+	if (str.charAt(0) === '-' ) {
+		
+		if (isNumber(str.charAt(1))) {
```

Comment:
Isn't this checking the same as in line 30 below?
Actually it probably doesn't work because of the `str[0]` below (which needs to change between 0 or 1 depending on the minus sign.